src/sprite_data.cpp

```diff
@@ -74,45 +73,43 @@ bool ImageData::Load8bpp(RcdFileReader *rcd_file, size_t length)
 
 		uint32 xpos = 0;
 		for (;;) {
-			if (offset + 2 >= length) return false;
+			if (offset + 2 >= length) rcd_file->Error("Offset out of bounds");
 			uint8 rel_pos = this->data[offset];
 			uint8 count = this->data[offset + 1];
 			xpos += (rel_pos & 127) + count;
 			offset += 2 + count;
 			if ((rel_pos & 128) == 0) {
-				if (xpos >= this->width || offset >= length) return false;
+				if (xpos >= this->width || offset >= length) rcd_file->Error("X coordinate out of inclusive bounds");
 			} else {
-				if (xpos > this->width || offset > length) return false;
+				if (xpos > this->width || offset > length) rcd_file->Error("X coordinate out of exclusive bounds");
```

Comment:
Is `inclusive` and `exclusive` correct at lines 82 and 84?

If boundary values are allowed (ie inclusive, right?), then you wouldn't throw an error on `==`